src/diet.ts

```diff
@@ -147,140 +185,49 @@ function getBestSpleenItems() {
     const synthTurns = haveEffect($effect`Synthesis: Greed`);
     const spleenTotal = spleenLimit() - mySpleenUse();
-    const adventuresPerItem = adventureGain(spleenItem);
+    const adventuresPerSpleen = getAverageAdventures(spleenItem) / spleenItem.spleen;
     // when not barfing, only get synth for estimatedTurns() turns (ignore adv gain)
-    const spleenAdvsGained = globalOptions.noBarf ? 0 : 1.04 * adventuresPerItem * spleenTotal;
+    const spleenAdvsGained = globalOptions.noBarf ? 0 : 1.04 * adventuresPerSpleen * spleenTotal;
     const spleenSynth = Math.ceil(
-      (spleenAdvsGained + estimatedTurns() - synthTurns) / (30 + 1.04 * adventuresPerItem)
+      (spleenAdvsGained + estimatedTurns() - synthTurns) / (30 + 1.04 * adventuresPerSpleen)
     );
     if (have($skill`Sweet Synthesis`)) {
-      for (let i = 0; i < clamp(spleenSynth, 0, spleenLimit() - mySpleenUse()); i++) {
-        sweetSynthesis($effect`Synthesis: Greed`);
-      }
+      sweetSynthesis(
+        clamp(spleenSynth, 0, spleenLimit() - mySpleenUse()),
+        $effect`Synthesis: Greed`
+      );
     }
-    const count = Math.floor((spleenLimit() - mySpleenUse()) / spleenItem.spleen);
+    const count = Math.max(0, Math.floor((spleenLimit() - mySpleenUse()) / spleenItem.spleen));
     chewSafe(count, spleenItem);
   }
 }
 
-function fillStomach() {
-  if (myLevel() >= 15 && !get("_hungerSauceUsed") && mallPrice($item`Hunger™ Sauce`) < 3 * MPA) {
-    acquire(1, $item`Hunger™ Sauce`, 3 * MPA);
-    use(1, $item`Hunger™ Sauce`);
-  }
-  useIfUnused($item`milk of magnesium`, "_milkOfMagnesiumUsed", 5 * MPA);
-
-  while (myFullness() + 5 <= fullnessLimit()) {
-    if (have($item`Universal Seasoning`) && !get("_universalSeasoningUsed")) {
-      use($item`Universal Seasoning`);
-    }
-    if (myMaxhp() < 1000) maximize("0.05hp, hot res", false);
-    const count = Math.floor(Math.min((fullnessLimit() - myFullness()) / 5, mySpleenUse() / 5));
-    if (mallPrice(saladFork) < (55 * MPA) / 6) {
-      acquire(count, saladFork, (55 * MPA) / 6, false);
-      eat(Math.min(count, itemAmount(saladFork)), saladFork);
-    }
-    MayoClinic.setMayoMinder(MayoClinic.Mayo.flex, count);
-    eatSpleen(count, $item`extra-greasy slider`);
-    fillSomeSpleen();
-  }
-}
-
-function fillLiverAstralPilsner() {
-  if (have($item`astral six-pack`)) {
-    use($item`astral six-pack`);
-  }
-  if (availableAmount($item`astral pilsner`) === 0) {
-    return;
-  }
-  try {
-    if (
-      !get("_mimeArmyShotglassUsed") &&
-      itemAmount($item`mime army shotglass`) > 0 &&
-      availableAmount($item`astral pilsner`) > 0
-    ) {
+function fillShotglass() {
+  if (!get("_mimeArmyShotglassUsed") && have($item`mime army shotglass`)) {
+    if (have($item`astral pilsner`)) {
       drinkSafe(1, $item`astral pilsner`);
+    } else {
+      equip($item`tuxedo shirt`);
+      drinkSafe(1, $item`splendid martini`);
```

Comment:
is the martini pretty much always going to be our best bet? Do we envision a scenario where a turkey booze or a sacramento wins out here?

Reply:
probably, but leaving outside scope for now